File: tests/Generator/resource/java/complex/GenericPropertyType.java
package org.typeapi.model;

import com.fasterxml.jackson.annotation.*;

@JsonClassDescription("Represents a generic value which can be replaced with a concrete type")
public class GenericPropertyType extends PropertyType {
    @JsonPropertyDescription("The name of the generic, it is recommended to use common generic names like T or TValue. These generics can then be replaced on usage with a concrete type through the template property at a reference")
    private String name;
    private String type = "generic";

    @JsonSetter("name")
    public void setName(String name) {
        this.name = name;
    }

    @JsonGetter("name")
    public String getName() {
        return this.name;
    }

    @JsonSetter("type")
    public void setType(String type) {
        this.type = type;
    }

    @JsonGetter("type")
    public String getType() {
        return this.type;
    }
}

File: tests/Generator/resource/kotlin/complex/GenericPropertyType.kt

import com.fasterxml.jackson.annotation.*

/**
 * Represents a generic value which can be replaced with a concrete type
 */
open class GenericPropertyType : PropertyType {
    @JsonProperty("name")
    var name: String? = null
    @JsonProperty("type")
    var type: String? = "generic"
}

